ports/nrf/common-hal/_bleio/Adapter.c

```diff
@@ -78,34 +89,56 @@ STATIC uint32_t ble_stack_enable(void) {
     // Start with no event handlers, etc.
     ble_drv_reset();
 
-    uint32_t app_ram_start;
-    app_ram_start = 0x20004000;
+    // Set everything up to have one persistent code editing connection and one user managed
+    // connection. In the future we could move .data and .bss to the other side of the stack and
+    // dynamically adjust for different memory requirements of the SD based on boot.py
+    // configuration.
+    uint32_t app_ram_start = (uint32_t) &_ram_start;
 
     ble_cfg_t ble_conf;
     ble_conf.conn_cfg.conn_cfg_tag = BLE_CONN_CFG_TAG_CUSTOM;
-    ble_conf.conn_cfg.params.gap_conn_cfg.conn_count = BLE_GAP_CONN_COUNT_DEFAULT;
+    ble_conf.conn_cfg.params.gap_conn_cfg.conn_count = BLEIO_TOTAL_CONNECTION_COUNT;
+    // Event length here can influence throughput so perhaps make multiple connection profiles
+    // available.
     ble_conf.conn_cfg.params.gap_conn_cfg.event_length = BLE_GAP_EVENT_LENGTH_DEFAULT;
     err_code = sd_ble_cfg_set(BLE_CONN_CFG_GAP, &ble_conf, app_ram_start);
-    if (err_code != NRF_SUCCESS)
+    if (err_code != NRF_SUCCESS) {
         return err_code;
+    }
 
     memset(&ble_conf, 0, sizeof(ble_conf));
-    ble_conf.gap_cfg.role_count_cfg.periph_role_count = 1;
+    ble_conf.gap_cfg.role_count_cfg.adv_set_count = 1;
+    ble_conf.gap_cfg.role_count_cfg.periph_role_count = 2;
     ble_conf.gap_cfg.role_count_cfg.central_role_count = 1;
     err_code = sd_ble_cfg_set(BLE_GAP_CFG_ROLE_COUNT, &ble_conf, app_ram_start);
-    if (err_code != NRF_SUCCESS)
+    if (err_code != NRF_SUCCESS) {
         return err_code;
+    }
 
     memset(&ble_conf, 0, sizeof(ble_conf));
     ble_conf.conn_cfg.conn_cfg_tag = BLE_CONN_CFG_TAG_CUSTOM;
     ble_conf.conn_cfg.params.gatts_conn_cfg.hvn_tx_queue_size = MAX_TX_IN_PROGRESS;
     err_code = sd_ble_cfg_set(BLE_CONN_CFG_GATTS, &ble_conf, app_ram_start);
-    if (err_code != NRF_SUCCESS)
+    if (err_code != NRF_SUCCESS) {
         return err_code;
+    }
 
+    // Double the GATT Server attribute size to accomodate both the CircuitPython built-in service
+    // and anything the user does.
+    memset(&ble_conf, 0, sizeof(ble_conf));
+    ble_conf.gatts_cfg.attr_tab_size.attr_tab_size = BLE_GATTS_ATTR_TAB_SIZE_DEFAULT * 3;
```

Comment:
Changed to triple. It's all a rough guess. It isn't connection related because this is the attribute table which holds service, characteristic and descriptor info I believe. When it's too small creating a Service or similar will fail with NOMEM.